### PR TITLE
fix(metrics): Implement accurate histogram zooming [INGEST-947]

### PR DESCRIPTION
**Pull request by untitaker**
Implement histogram zooming based on bucket filtering. Together with outlier tagging done in Relay the story should finally come to an end.

### CODE REVIEW

File: src/sentry/snuba/metrics/fields/base.py [hunk 1]
```diff
@@ -457,12 +487,22 @@ def generate_bottom_up_derived_metrics_dependencies(
...
+            org_id=org_id, query_definition=query_definition
+        )
+        if operation_based_filter is not None:
+            rv = Function("and", [rv, operation_based_filter])
+
         return Function(
             snuba_function,
-            [Column("value"), self.metric_object.generate_filter_snql_conditions(org_id=org_id)],
-            alias=f"{self.metric_operation.op}({self.metric_object.metric_mri})",
+            [Column("value"), rv],
```
Comment (ahmedetefy) [inline src/sentry/snuba/metrics/fields/base.py:504]: nit: Can you please name this something other than rv?

File: src/sentry/snuba/metrics/fields/base.py [hunk 2]
```diff
@@ -457,12 +487,22 @@ def generate_bottom_up_derived_metrics_dependencies(
...
+            f"{self.metric_operation.op}({self.metric_object.metric_mri})",
```
Comment (ahmedetefy) [inline src/sentry/snuba/metrics/fields/base.py:505]: Why did you remove `alias=`?

**Review by ahmedetefy — APPROVED** [review]

File: src/sentry/snuba/metrics/fields/base.py (hunk 2 above)
Comment (untitaker) [reply src/sentry/snuba/metrics/fields/base.py]: no reason, i moved code around and reverted things manually again
